File: java/stdin.java
// Test script to get input, transform, and write to stdout
import java.util.Scanner;

public class Main{
  public static void main(String[] args){
    int i = 1;
    Scanner scanner = new Scanner(System.in);
    while (scanner.hasNextLine()) {
        String userInput = scanner.nextLine();
        System.out.println(i + " " + userInput.toUpperCase());
        i++;
  }
}
}

File: lua/stdin.lua
-- Lua script to read input, transform to uppercase, and print with line numbers
local i = 1

-- Read from stdin until EOF
for user_input in io.lines() do
    -- Transform input to uppercase and prepend with line number
    print(i .. " " .. user_input:upper())
    i = i + 1
end


